tests/countries/test_equatorial_guinea.py

```diff
@@ -0,0 +1,171 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from unittest import TestCase
+
+from holidays.countries.equatorial_guinea import EquatorialGuinea, GQ, GNQ
+from tests.common import CommonCountryTests
+
+
+class TestEquatorialGuinea(CommonCountryTests, TestCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass(EquatorialGuinea)
+
+    def test_country_aliases(self):
+        self.assertAliases(EquatorialGuinea, GQ, GNQ)
+
+    def test_new_years_day(self):
+        name = "Año Nuevo"
+        self.assertHolidayName(name, (f"{year}-01-01" for year in range(1968, 2050)))
+
+        obs_dt = "2017-01-02"
+        self.assertHolidayName(f"{name} (observado)", obs_dt)
+        self.assertNoNonObservedHoliday(obs_dt)
+
+    def test_womens_day(self):
+        name = "Día Internacional de la Mujer"
+        self.assertHolidayName(name, (f"{year}-03-08" for year in range(1968, 2050)))
+
+    def test_good_friday(self):
+        name = "Viernes Santo"
+        self.assertHolidayName(
+            name,
+            "1994-04-01",
+            "1995-04-14",
+            "1996-04-05",
+            "1997-03-28",
+            "2017-04-14",
+            "2018-03-30",
+            "2019-04-19",
+            "2020-04-10",
+        )
+
+    def test_labor_day(self):
+        name = "Día del Trabajo"
+        self.assertHolidayName(name, (f"{year}-05-01" for year in range(1968, 2050)))
+
+        obs_dt = ("2021-04-30", "2016-05-02", "2022-05-02")
+        self.assertHolidayName(f"{name} (observado)", obs_dt)
+        self.assertNoNonObservedHoliday(obs_dt)
+
+    def test_corpus_christi_day(self):
+        name = "Corpus Christi"
+        self.assertHolidayName(
+            name,
+            "2020-06-11",
+            "2021-06-03",
+            "2022-06-16",
+            "2023-06-08",
+            "2024-05-30",
+            "2025-06-19",
+            "2026-06-04",
+            "2027-05-27",
+            "2028-06-15",
+            "2029-05-31",
+            "2030-06-20",
+        )
+
+    def test_armed_forces_day(self):
+        name = "Día de las Fuerzas Armadas"
+        self.assertHolidayName(name, (f"{year}-08-03" for year in range(1979, 2050)))
+
```

Comment:
For holidays entries like this, `self.assertNoHolidayName( ... )`(s) for years with no observances before and/or after the `start_year` are needed as well:
```suggestion
    def test_armed_forces_day(self):
        name = "Día de las Fuerzas Armadas"
        self.assertHolidayName(name, (f"{year}-08-03" for year in range(1979, 2050)))
        self.assertNoHolidayName(name, range(1969, 1979))

```